app/src/sharedTest/java/org/oppia/app/home/HomeActivityTest.kt

```diff
@@ -89,10 +99,185 @@ class HomeActivityTest {
+  fun testHomeActivity_recyclerViewIndex0_displaysWelcomeMessageCorrectly() {
+    launch(HomeActivity::class.java).use {
+      onView(
+        atPositionOnView(
+          R.id.home_recycler_view,
+          0,
+          R.id.welcome_text_view
+        )
+      ).check(matches(withText(containsString("Welcome"))))
+    }
+  }
+
+  @Test
+  fun testHomeActivity_recyclerViewIndex0_configurationChange_displaysWelcomeMessageCorrectly() {
+    launch(HomeActivity::class.java).use {
+      it.onActivity { activity ->
+        activity.requestedOrientation = Configuration.ORIENTATION_LANDSCAPE
+      }
+      onView(
+        atPositionOnView(
+          R.id.home_recycler_view,
+          0,
+          R.id.welcome_text_view
+        )
+      ).check(matches(withText(containsString("Welcome"))))
+    }
+  }
+
+  @Test
+  fun testHomeActivity_recyclerViewIndex1_displaysRecentlyPlayedStoriesText() {
+    launch(HomeActivity::class.java).use {
+      onView(atPositionOnView(R.id.home_recycler_view, 1, R.id.recently_played_stories_text_view)).check(
+        matches(
+          withText(R.string.recently_played_stories)
+        )
+      )
+    }
+  }
+
+  @Test
+  fun testHomeActivity_recyclerViewIndex1_displaysViewAllText() {
+    launch(HomeActivity::class.java).use {
+      onView(atPositionOnView(R.id.home_recycler_view, 1, R.id.view_all)).check(
+        matches(
+          withText(R.string.view_all)
+        )
+      )
+    }
+  }
+
+  @Test
+  fun testHomeActivity_recyclerViewIndex1_clickViewAll_opensContinuePlayingActivity() {
+    launch(HomeActivity::class.java).use {
+      onView(atPositionOnView(R.id.home_recycler_view, 1, R.id.view_all)).perform(click())
+      intended(hasComponent(ContinuePlayingActivity::class.java.name))
+    }
+  }
+
+  @Test
+  fun testHomeActivity_recyclerViewIndex1_promotedCard_chapterNameIsCorrect() {
+    launch(HomeActivity::class.java).use {
+      onView(atPositionOnView(R.id.home_recycler_view, 1, R.id.chapter_name)).check(
+        matches(
+          withText(containsString("The Meaning of Equal Parts"))
+        )
+      )
+    }
+  }
+
+  @Test
+  fun testHomeActivity_recyclerViewIndex1_promotedCard_storyNameIsCorrect() {
+    launch(HomeActivity::class.java).use {
+      onView(atPositionOnView(R.id.home_recycler_view, 1, R.id.story_name)).check(
+        matches(
+          withText(containsString("Second Story"))
+        )
+      )
+    }
+  }
+
+  @Test
+  fun testHomeActivity_recyclerViewIndex1_configurationChange_promotedCard_storyNameIsCorrect() {
+    launch(HomeActivity::class.java).use {
+      it.onActivity { activity ->
+        activity.requestedOrientation = Configuration.ORIENTATION_LANDSCAPE
+      }
+      onView(atPositionOnView(R.id.home_recycler_view, 1, R.id.story_name)).check(
+        matches(
+          withText(containsString("Second Story"))
+        )
+      )
+    }
+  }
+
+  // TODO(#135): Update this test case to check that topic-play tab is opened.
```

Comment:
@nikitamarysolomanpvt FYI.

Reply:
Also @rt4914, shouldn't we test this within the topic activity/fragment test, instead? Here it makes sense to me to verify we're using the correct API (sending a topic & story ID), and have another test for topic activity to verify that starting the activity with both parameters properly auto opens the tab & scrolls.

If you agree, let's finish the home fragment test side here and move this TODO to a checkbox within #135 that can be satisfied with a test in TopicActivityTest.

Reply:
Done. I have made changes to the description in #135 and your approach makes much more sense to me.